PowerEditor/src/ScitillaComponent/FindReplaceDlg.cpp

```diff
@@ -2158,15 +2159,15 @@ int FindReplaceDlg::processRange(ProcessOperation op, FindReplaceInfo & findRepl
 				int nbChar = lend - lstart;
 
 				// use the static buffer
-				TCHAR lineBuf[1024];
+				TCHAR lineBuf[SC_SEARCHRESULT_LINEBUFFERMAXLENGTH];
 
-				if (nbChar > 1024 - 3)
-					lend = lstart + 1020;
+				if (nbChar > SC_SEARCHRESULT_LINEBUFFERMAXLENGTH - 3)
+					lend = lstart + SC_SEARCHRESULT_LINEBUFFERMAXLENGTH - 4;

```

Comment:
@Uhf7 
You eliminated the `1024` "magic number" (good!).  
How about doing something better for the `3` and the `4` magic numbers as well?
I'm fairly certain that `3` refers to the length of `...`, but that's because I'm familiar with this code (future code readers may not know this).
Perhaps the `4` should not be used anyway -> it should be `- 3 - 1` (before `3` is eliminated)

Reply:
Actually, I'm not even sure if there should be a new constant for this. Reason: The limit in wide chars determines, how many characters we want to see in the Find Result window. The `SC_SEARCHRESULT_LINEBUFFERMAXLENGTH` constant says, how many UTF-8 characters the parser can handle. And between the new constant and the `SC_SEARCHRESULT_LINEBUFFERMAXLENGTH`  constant should be the factor 4.